### ISSUE TITLE
Add "permission" to the setting property

### ISSUE THREAD

**Issue by wakuflair** [EasyAbp/Abp.SettingUi#43]
see https://github.com/EasyAbp/Abp.SettingUi/issues/40#issuecomment-729326423

**Comment by cihangll** [comment]
Hi @wakuflair,
I think we don't need `Permission` field. We already know the group and setting name at runtime. We can implement this as default behaviour. If user define permission like SettingUiPermissions.Tenant.**GroupName**.**SettingName** we can automatically check this at runtime. If user dont define this permission we can pass and show specified setting.

For example, we have 2 different groups and 4 different setting for each group. We can write like this. (spaghetti code)

```csharp
var moduleGroup = context.AddGroup(SettingUiPermissions.GroupName, L("Permission:SettingUi"));
var root = moduleGroup.AddPermission(SettingUiPermissions.Tenant, L("Permission:SettingUi.Tenant"));

var group1 = root.AddChild(SettingUiPermissions.Tenant + ".Group1", L("Permission:SettingUi.Tenant.Group1"));
group1.AddChild(SettingUiPermissions.Tenant + ".Group1" + ".Setting1"); // If the user has this permission, show the specified setting
//group1.AddChild(SettingUiPermissions.Tenant + ".Group1" + ".Setting2"); always show
//group1.AddChild(SettingUiPermissions.Tenant + ".Group1" + ".Setting3"); always show
group1.AddChild(SettingUiPermissions.Tenant + ".Group1" + ".Setting4"); // If the user has this permission, show the specified setting

var group2 = root.AddChild(SettingUiPermissions.Tenant + ".Group2", L("Permission:SettingUi.Tenant.Group2"));
//group2.AddChild(SettingUiPermissions.Tenant + ".Group2" + ".Setting1"); always show
//group2.AddChild(SettingUiPermissions.Tenant + ".Group2" + ".Setting2"); always show
//group2.AddChild(SettingUiPermissions.Tenant + ".Group2" + ".Setting3"); always show
group2.AddChild(SettingUiPermissions.Tenant + ".Group2" + ".Setting4");// If the user has this permission, show the specified setting
```
![image](https://user-images.githubusercontent.com/6229029/102699774-6c82d180-4258-11eb-9be8-ab155ba2f756.png)

**Comment by wakuflair** [comment]
Sounds a good idea!

I will take a closer look, thank you @cihangll :smile:

**Comment by cihangll** [comment]
I can PR for this. :)

**Comment by cihangll** [comment]
Hi @wakuflair ,
Can you check [this](https://github.com/EasyAbp/Abp.SettingUi/pull/46) PR for this?

**Comment by wakuflair** [comment]
Sure, I will look at it ASAP.

**Comment by wakuflair** [comment]
Resolved with #46
